app/controllers/reports_controller.rb

```diff
@@ -0,0 +1,45 @@
+require 'report_email_sender'
+
+class ReportsController < ApplicationController
+  def new
+    @report = Report.new
+  end
+
+  def create
+    @report = Report.generate_report_for_keywords(params['keywords'].values)
+    if @report.save
+      render action: 'select_soc_codes', guid: @report.guid
+    else
+      @error = 'Please enter at least 1 job or sector.'
+      render action: 'new'
+    end
+  end
+
+  def show
+    @report = Report.where(guid: params[:id]).first
+  end
+
+  def save_soc_codes
+    @report = Report.where(guid: params[:id]).first
```

Comment:
oooh, or we could do this 
```ruby
Report.friendly.find(params[:id])
```
(from here: https://github.com/norman/friendly_id#rails-quickstart)